pkg/api/mock/matcher/matcher_test.go

```diff
@@ -146,13 +145,6 @@ var _ = Describe("custom gomega matcher", func() {
 			Expect(a.All()).To(ContainElementNTimes(Object(&testObject2{}), 1))
 		})
 
-		It("returns error when match is called with uncomparable type", func() {
```

Comment:
I think we can still have cases of `interface Object`-valid Objects that are incompatible with `pkg/utils/object/compare` and still need some checking for handling them correctly - so this case still makes sense I think